### PR TITLE
CDAP-3727 fix bug with order of system artifact loading

### PR DESCRIPTION
**Pull request by albertshau**
The previous implementation tried to enforce ordering by sorting
on the artifacts, with parents ordered larger than children. The
problem was the comparison make sense when 2 artifacts were being
compared, but it wasn't a natural ordering. That is, there could
be a scenario where A < B, B < C, but A > C.

Instead, since we enforce only 1 level of dependencies, just
adding all parents first, then adding the children.

### CODE REVIEW

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -187,7 +187,8 @@
 
   <property>
     <name>app.artifact.dir</name>
-    <value>/opt/cdap/master/artifacts</value>
+    <!--value>/opt/cdap/master/artifacts</value-->
```
Comment (gokulavasan) [inline cdap-common/src/main/resources/cdap-default.xml]: Is this change intentional? If yes, should we retain the comment?

Comment (albertshau) [inline cdap-common/src/main/resources/cdap-default.xml]: ugh no it's not, changing it back

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-RBT472

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepositoryTest.java [hunk 2]
```diff
@@ -141,10 +141,10 @@ public void testAddSystemArtifacts() throws Exception {
       createManifest(ManifestFields.EXPORT_PACKAGE, PluginTestRunnable.class.getPackage().getName()));
 
     // write plugins jar
-    Id.Artifact pluginArtifactId = Id.Artifact.from(Id.Namespace.SYSTEM, "myPlugin", "1.0.0");
+    Id.Artifact pluginArtifactId = Id.Artifact.from(Id.Namespace.SYSTEM, "APlugin", "1.0.0");
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepositoryTest.java]: Do the renaming in this test test the fix? If yes, how?

**Comment by gokulavasan** [comment]
One question. Rest LGTM :+1:

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepositoryTest.java (hunk 2 above)
Comment (albertshau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepositoryTest.java]: o this was leftovers from debugging.  I will modify the test to actually test it... need to add another plugin jar to this test case.
